src/repository/mod.rs

```diff
@@ -43,16 +50,138 @@ fn create_db(con: &mut Connection) {
 /// If not, it either creates or upgrades the database accordingly
 pub fn initialize_db() -> Result<()> {
     let mut con = open_connection();
+    let mut should_gen_database_from_files = false;
     // table_version will be used once we have more versions of the database
     let table_version = match metadata_repository::get_version(&con) {
         Ok(value) => value.parse::<u64>().unwrap(),
         Err(_) => {
             // tables haven't been created yet
             create_db(&mut con);
+            should_gen_database_from_files = true;
             1
         }
     };
     migrate_db(&con, table_version)?;
+    if should_gen_database_from_files {
+        generate_database_from_files(None, &con)?;
+    }
     con.close().unwrap();
     Ok(())
 }
+
+/// Generates database entries from the existing files directory structure.
+/// This walks the directory tree depth-first, creating folders before files at each level.
+///
+/// This function is designed to be called with `parent_folder = None` to start the
+/// generation from the root files directory. The `parent_folder` parameter exists
+/// to satisfy the API contract but the actual recursive traversal is handled internally.
+///
+/// # Arguments
+/// * `parent_folder` - Should be None to start from root. Any other value is a no-op.
+/// * `con` - Database connection
+///
+/// # Returns
+/// * `Result<()>` - Ok if successful, or a rusqlite error
+pub fn generate_database_from_files(parent_folder: Option<u32>, con: &Connection) -> Result<()> {
```

Comment:
this function takes a while on high-end hardware. Add info logging on status. Logging format should be

- when starting job: `Generating database from file system` (magenta text)
- when a folder is started: `Starting folder {name}` (grey text)
- when a folder is finished: `Finished folder {name}` (cyan text)
- When the entire job is done: `Finished generating database from file system` (green text)